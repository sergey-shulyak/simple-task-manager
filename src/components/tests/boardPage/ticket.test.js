import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Ticket from '../../boardPage/board/column/ticket';

const props = {
  id: '42',
  title: 'Sample title',
  description: 'Sample description',
  priority: 'medium',
  type: 'task',
  boardId: '24',
  className: 'custom'
};

describe('Ticket', () => {
  it('Should render correctly', () => {
    const wrapper = shallow(<Ticket {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
