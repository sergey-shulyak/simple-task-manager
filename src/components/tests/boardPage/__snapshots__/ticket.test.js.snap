// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket Should render correctly 1`] = `
<Link
  className="ticket"
  replace={false}
  to="/boards/24/tickets/42"
>
  <div
    className="ticket__content custom"
  >
    <h3
      className="ticket__title"
    >
      Sample title
    </h3>
    <p
      className="ticket__description"
    >
      Sample description
    </p>
    <p>
      <span
        className="ticket__field"
      >
        Type: 
      </span>
      <span
        className="ticket__icon"
      >
        <i
          aria-hidden="true"
          className="fa fa-wrench"
        />
         
      </span>
      <span
        className="ticket__type ticket__type_task"
      >
        task
      </span>
    </p>
    <p>
      <span
        className="ticket__field"
      >
        Priority: 
      </span>
      <span
        className="ticket__icon"
      >
        <i
          aria-hidden="true"
          className="fa fa-angle-up"
        />
         
      </span>
      <span
        className="ticket__priority ticket__priority_medium"
      >
        medium
      </span>
    </p>
  </div>
</Link>
`;
